task.py

```diff
@@ -435,7 +436,28 @@ def aggregate_output(self, out: tftbase.TftAggregateOutput) -> None:
             # This output has nothing to collect. We are done.
             return
 
-        self._aggregate_output(self._result, out)
+        result = self._result
+
+        if isinstance(result, tftbase.IperfOutput):
+            if out.flow_test is not None:
+                raise RuntimeError("There can only be one IperfOutput")
+            out.flow_test = result
+            if result.success:
+                log_level = logging.INFO
+                log_msg = "success"
+            else:
+                log_level = logging.ERROR
+                log_msg = "failure"
+            logger.log(log_level, f"Results of {self.ts.get_test_str()}: {log_msg}")
+            logger.debug(f"result: {common.dataclass_to_json(result)}")
+
+            if type(self)._aggregate_output is Task._aggregate_output:
+                # This instance did not overwrite _aggregate_output(). This is
+                # fine for a task that returned the IperfOutput. Don't call
+                # _aggregate_output.
+                return
+
+        self._aggregate_output(result, out)
```

Comment:
Wouldn't this fail now since we removed this method from the classes derived from Task?

Reply:
It's correct. See the lines above.

We have tests (`testType*py`) and plugins (`plugin*.py`). The former return `IperfOutput`, the latter return `PluginOutput` (both are `AggregatableOutput`).

Tasks that return an `AggregatableOutput` (either `IperfOutput` or `PluginOutput`) MUST overwrite `_aggreate_output()`, ... unless the task is a test (which returns `IperfOutput`), then it may choose not overwrite `_aggregate_output()`. The reason is to provide the convenience to not add code for the case where the test-task has nothing to aggregate further.

Reply:
extended code comment about that.

Reply:
@SalDaniele to be clear. This "convenience" to allow the test to not implement `_aggregate_output()` comes at the cost of requiring a few lines of code and several lines of code comments explaining why it's done that way. Additionally, surprising the reviewer.

Arguably, one might ask, wouldn't it be simpler to just require every implementation to just implement an empty `_aggregare_output()`? Maybe. But note that in the future we might have more test implementations, at which point this convenience starts to matter more.

This is not a very convincing argument. So... yeah. I just did it that way, for not very good reasons. It could also be different...